core/src/kvs/tx.rs

```diff
@@ -1,3170 +1,1520 @@
-where
-	K: Into<Key> + Debug,
-{
-	pub range: Range<K>,
-	pub limit: Limit,
-}
-
-impl From<Range<Vec<u8>>> for ScanPage<Vec<u8>> {
-	fn from(value: Range<Vec<u8>>) -> Self {
-		ScanPage {
-			range: value,
-			limit: Limit::Unlimited,
-		}
-	}
-}
-
-#[non_exhaustive]
-pub struct ScanResult<K>
-where
-	K: Into<Key> + Debug,
-{
-	pub next_page: Option<ScanPage<K>>,
-	pub values: Vec<(Key, Val)>,
-}
+const TARGET: &str = "surrealdb::core::kvs::tx";
 
-/// A set of undoable updates and requests against a dataset.
-#[allow(dead_code)]
 #[non_exhaustive]
 pub struct Transaction {
-	pub(super) inner: Inner,
-	pub(super) cache: Cache,
-	pub(super) cf: cf::Writer,
-	pub(super) vso: Arc<Mutex<Oracle>>,
-	pub(super) clock: Arc<SizedClock>,
-	pub(super) prepared_async_events: (Arc<Sender<TrackedResult>>, Arc<Receiver<TrackedResult>>),
-	pub(super) engine_options: EngineOptions,
-}
-
-#[allow(clippy::large_enum_variant)]
-pub(super) enum Inner {
-	#[cfg(feature = "kv-mem")]
-	Mem(super::mem::Transaction),
-	#[cfg(feature = "kv-rocksdb")]
-	RocksDB(super::rocksdb::Transaction),
-	#[cfg(feature = "kv-indxdb")]
-	IndxDB(super::indxdb::Transaction),
-	#[cfg(feature = "kv-tikv")]
-	TiKV(super::tikv::Transaction),
-	#[cfg(feature = "kv-fdb")]
-	FoundationDB(super::fdb::Transaction),
-	#[cfg(feature = "kv-surrealkv")]
-	SurrealKV(super::surrealkv::Transaction),
-}
-
-#[derive(Copy, Clone)]
-#[non_exhaustive]
-pub enum TransactionType {
-	Read,
-	Write,
-}
-
-impl From<bool> for TransactionType {
-	fn from(value: bool) -> Self {
-		match value {
-			true => TransactionType::Write,
-			false => TransactionType::Read,
-		}
-	}
-}
-
-#[non_exhaustive]
-pub enum LockType {
-	Pessimistic,
-	Optimistic,
-}
-
-impl fmt::Display for Transaction {
-	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-		#![allow(unused_variables)]
-		match &self.inner {
-			#[cfg(feature = "kv-mem")]
-			Inner::Mem(_) => write!(f, "memory"),
-			#[cfg(feature = "kv-rocksdb")]
-			Inner::RocksDB(_) => write!(f, "rocksdb"),
-			#[cfg(feature = "kv-indxdb")]
-			Inner::IndxDB(_) => write!(f, "indxdb"),
-			#[cfg(feature = "kv-tikv")]
-			Inner::TiKV(_) => write!(f, "tikv"),
-			#[cfg(feature = "kv-fdb")]
-			Inner::FoundationDB(_) => write!(f, "fdb"),
-			#[cfg(feature = "kv-surrealkv")]
-			Inner::SurrealKV(_) => write!(f, "surrealkv"),
-			#[allow(unreachable_patterns)]
-			_ => unreachable!(),
-		}
-	}
+	/// The underlying transactor
+	tx: Mutex<Transactor>,
+	/// The query cache for this store
+	cache: Cache<Key, Entry, EntryWeighter>,
 }
 
 impl Transaction {
-	// --------------------------------------------------
-	// Configuration methods
-	// --------------------------------------------------
-
-	pub fn rollback_with_warning(mut self) -> Self {
-		self.check_level(Check::Warn);
-		self
+	/// Create a new query store
+	pub fn new(tx: Transactor) -> Transaction {
+		Transaction {
+			tx: Mutex::new(tx),
+			cache: Cache::with_weighter(10_000, 10_000, EntryWeighter),
```

Comment:
Shouldn't we use a variable environment to set the size of the cache, and use 10_000 as a default value

Reply:
Good idea, I'll make the change in an upcoming commit!

Reply:
Fixed in https://github.com/surrealdb/surrealdb/pull/4257/commits/0ee20c671f1473ad33772a4e45cf906d4f27eab1